### PR TITLE
add test for velocity verlet

### PR DESCRIPTION
**Pull request by rlouf**
This PR add tests for the velocity verlet algorithm implementation

### CODE REVIEW

File: tests/test_integrators.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    p = aet.vector("p")
+    step_size = aet.scalar("step_size")
+    energy = potential(q)
+    energy_grad = aesara.grad(energy, q)
+    trajectory, _ = aesara.scan(
+        fn=step_fn,
+        outputs_info=[q, p, energy, energy_grad],
+        non_sequences=step_size,
+        n_steps=n_steps,
+    )
```
Comment (rlouf) [inline tests/test_integrators.py:67]: This fails with the following error message:

```
ValueError: Scan cannot parse the return value of your lambda expression, which is: (Elemwise{add,no_inplace}.0, Elemwise{sub,no_inplace}.0, Sum{acc_dtype=float64}.0, [Elemwise{mul}.0])
```

and I cannot figure out why after reading the documentation and the code.

**Review by brandonwillard — CHANGES_REQUESTED** [review]
This change breaks `mypy` checking.  Did you set up `pre-commit`, because it should've prevented this commit?

**Comment by brandonwillard** [comment]
@rlouf, I've added a commit that adjust the use of `aesara.function` and `aesara.scan` so that the evaluations can be performed.

**Comment by rlouf** [comment]
@brandonwillard I fixed the few bugs that were left, added tests to check that we integrate the correct trajectory, and a test on 2D state vectors. Good on my end.

**Pull request merged**